static/app/views/insights/pages/backend/laravelOverviewPage.tsx

```diff
@@ -636,6 +638,185 @@ function JobsWidget({query}: {query?: string}) {
   );
 }
 
+interface QueriesDiscoverQueryResponse {
+  data: Array<{
+    'avg(span.self_time)': number;
+    'project.id': string;
+    'span.description': string;
+    'span.group': string;
+    'span.op': string;
+    'sum(span.self_time)': number;
+    'time_spent_percentage()': number;
+    transaction: string;
+  }>;
+}
+
+function QueriesWidget({query}: {query?: string}) {
+  const organization = useOrganization();
+  const pageFilterChartParams = usePageFilterChartParams();
+
+  const queriesRequest = useApiQuery<QueriesDiscoverQueryResponse>(
+    [
+      `/organizations/${organization.slug}/events/`,
+      {
+        query: {
+          ...pageFilterChartParams,
+          dataset: 'spansMetrics',
+          field: [
+            'span.op',
+            'span.group',
+            'project.id',
+            'span.description',
+            'sum(span.self_time)',
+            'avg(span.self_time)',
+            'time_spent_percentage()',
+            'transaction',
+          ],
+          query: `has:span.description span.module:db ${query}`,
+          referrer: 'api.performance.generic-widget-chart.most-time-spent-db-queries',
+          sort: '-time_spent_percentage()',
+          per_page: 3,
+        },
+      },
+    ],
+    {staleTime: 0}
+  );
+
+  const timeSeriesRequest = useApiQuery<MultiSeriesEventsStats>(
+    [
+      `/organizations/${organization.slug}/events-stats/`,
+      {
+        query: {
+          ...pageFilterChartParams,
+          dataset: 'spansMetrics',
+          field: ['span.group', 'sum(span.self_time)'],
+          yAxis: ['sum(span.self_time)'],
+          query: `span.group:[${queriesRequest.data?.data.map(item => item['span.group']).join(',')}]`,
+          orderby: '-sum(span.self_time)',
+          topEvents: 3,
+        },
+      },
+    ],
+    {
+      staleTime: 0,
+      enabled: !!queriesRequest.data?.data,
+    }
+  );
```

Comment:
You might have an easier time if you use the `useDiscoverSeries` data hook and its neighbours. e.g., `useSpanMetricsSeries` might be helpful, it returns `DiscoverSeries` objects that you can pass directly to `InsightsBarChartWidget`, or use the `convertSeriesToTimeseries` helper to convert `DiscoverSeries` to `TimeSeries`

FWIW it's not lost on me how annoying this is! I hope I have time to improve this situation soon 🙏🏻

Reply:
Also, the `useDiscoverX` hooks are _typed_ so the response will have the correct type for whichever fields you requested!